File: fetch_cities.js
const fs = require('fs');
const path = require('path');

const DATA_DIR = path.join(__dirname, 'src', 'data');
if (!fs.existsSync(DATA_DIR)) {
  fs.mkdirSync(DATA_DIR, { recursive: true });
}

async function main() {
  try {
    const response = await fetch('https://turkiyeapi.dev/api/v1/provinces');
    const json = await response.json();
    
    const citiesMap = {};
    const provinces = json.data;
    
    for (const province of provinces) {
      const cityName = province.name;
      const districts = province.districts.map(d => d.name);
      citiesMap[cityName] = districts.sort((a, b) => a.localeCompare(b, 'tr'));
    }

    fs.writeFileSync(path.join(DATA_DIR, 'cities.json'), JSON.stringify(citiesMap, null, 2), 'utf8');
    console.log('cities.json created successfully.');
  } catch (e) {
    console.error("Error:", e);
  }
}

main();

File: fetch_cities.py
import urllib.request
import json
import os

url = "https://gist.githubusercontent.com/sercanov/c63063e4b40c756d4040a0be694895e9/raw"

try:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
    
    cities_map = {}
    for item in data:
        city_name = item['il']
        districts = [d['ilce'] for d in item['ilceler']]
        cities_map[city_name] = sorted(districts)

    os.makedirs('src/data', exist_ok=True)
    with open('src/data/cities.json', 'w', encoding='utf-8') as f:
        json.dump(cities_map, f, ensure_ascii=False, indent=2)
    print("Success")
except Exception as e:
    print("Error:", e)
